### ISSUE TITLE
[BUG] COMPOSE_PROFILES environment var no longer works across included compose files

### ISSUE THREAD

**Issue by badders** [docker/compose#10906]
### Description

Profiles configured in the COMPOSE_PROFILE environment variable are no longer started when running` docker compose up`

### Steps To Reproduce

1. export COMPOSE_PROFILES=profile1,profile2
2. docker compose up --build -d
3. Note that containers specified as belonging to profile1 or profile2 are _not_ started

### Compose Version

```Text
Docker Compose version v2.20.2-desktop.1
```

### Docker Environment

```Text
Client:
 Version:    24.0.5
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.1
    Path:     /Users/tom/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.20.2-desktop.1
    Path:     /Users/tom/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/tom/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     /Users/tom/.docker/cli-plugins/docker-extension
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.6
    Path:     /Users/tom/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/tom/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/tom/.docker/cli-plugins/docker-scan
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  0.20.0
    Path:     /Users/tom/.docker/cli-plugins/docker-scout

Server:
 Containers: 18
  Running: 7
  Paused: 0
  Stopped: 11
 Images: 31
 Server Version: 24.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 5.15.49-linuxkit-pr
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 7.773GiB
 Name: docker-desktop
 ID: 699641c6-1601-443f-afd8-cee881675044
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false
```

### Anything else?

Docs for COMPOSE_PROFILES https://docs.docker.com/compose/profiles/#start-specific-profiles

**Comment by badders** [comment]
Ah it seems to be tied to including files, heres an example of this:

`docker-compose.yml`

```
include:
-  more-services.yml

services:
  always:
   image: alpine
   command: ['sleep', 'infinity']
   init: true
```

`more-services.yml`

```
services:
  prof1:
   build: .
   profiles: ['profile1']
  prof2:
   build: .
   profiles: ['profile2']
```

1. With COMPOSE_PROFILES (doesnt work):

```
COMPOSE_PROFILES=profile1,profile2 docker compose up --build -d                                                                                                                                        ☸ grimmauld-place
[+] Running 2/2
 ✔ Network scratch_default     Created                                                                                                                                                                                                    0.1s
 ✔ Container scratch-always-1  Started
```

2. With profiles set on command line (works):

```
docker compose --profile profile1 --profile profile2 up --build -d                                                                                                                                     ☸ grimmauld-place
[+] Building 0.8s (8/8) FINISHED                                                                                                                                                                                          docker:desktop-linux
--- removed for brevity
[+] Running 3/0
 ✔ Container scratch-prof2-1   Running                                                                                                                                                                                                    0.0s
 ✔ Container scratch-always-1  Running                                                                                                                                                                                                    0.0s
 ✔ Container scratch-prof1-1   Running
```

**Comment by milas** [comment]
Ahhh yep, that was it! Thank you for the clear issue report.

I've got a fix @ compose-spec/compose-go#446. Will update this issue once it's released in Compose.

**Comment by rollsover** [comment]
+

**Comment by badders** [comment]
Fantastic, thanks @milas

**Comment by milas** [comment]
This is fixed in [v2.21.0](https://github.com/docker/compose/releases/tag/v2.21.0).

Thanks again for the report!
